Gateway/Validator/CheckoutResponseValidator.php

```diff
@@ -151,6 +151,10 @@ public function validate(array $validationSubject)
                         $paymentData = $response['paymentData'];
                     }
 
+                    if (!empty($response['details'])) {
+                        $details = $response['details'];
+                    }
```

Comment:
why doing this here ? I only see this being used in the else can we combine it with that if ?